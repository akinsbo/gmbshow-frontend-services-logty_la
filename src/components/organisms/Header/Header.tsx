import AppBar from '@material-ui/core/AppBar'
import IconButton from '@material-ui/core/IconButton'
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles'
import siteDataJson from "../../../utils/site-data.json"
import Toolbar from '@material-ui/core/Toolbar'
import Typography from '@material-ui/core/Typography'
// tslint:disable-next-line:import-name
import MenuIcon from '@material-ui/icons/Menu'
import * as React from 'react'

// create a styles object using a theme. The createStyles function is
// needed to placate the TS compiler. 
// See https://gist.github.com/johnsogg/dde852d76aa7af92c1cedd0e594557db
const useStyles = makeStyles((theme: Theme) => createStyles({
    grow: {
        flexGrow: 1,
    },
    menuButton: {
        marginRight: theme.spacing(2),
    },
    root: {
        width: '100%',
    },
    titleStyle: {
        display: 'none',
        [theme.breakpoints.up('sm')]: {
            display: 'block',
        },
    },
    input: {
        display: 'none',
    },
    sectionDesktop: {
        display: 'none',
        [theme.breakpoints.up('md')]: {
            display: 'flex',
        },
    },
    sectionMobile: {
        display: 'flex',
        [theme.breakpoints.up('md')]: {
            display: 'none',
        },
    },
})
)

interface HeaderProps {
    title?: string
    children?: JSX.Element[] | JSX.Element | string[] | string
}

// export interface HeaderProps extends WithStyles<typeof styles> { }
const Header = (props: HeaderProps) => {
    const classes = useStyles()
    const { grow, menuButton, titleStyle, sectionDesktop, sectionMobile } = classes

    return (
        <React.Fragment>
            <div className={`header ${grow}`}>
                <AppBar position="static">
                    <Toolbar>
                        <IconButton
                            edge="start"
                            className={menuButton}
                            color="inherit"
                            aria-label="Open drawer"
                        >
                            <MenuIcon />
                        </IconButton>
                        <Typography className={titleStyle} variant="h6" color="inherit" noWrap>
                            title
                         </Typography>
                        <div className={grow} />
                        <div className={sectionDesktop} />
                        <div className={sectionMobile} />
                    </Toolbar>
                </AppBar>
            </div>
        </React.Fragment>
    )
}
const {title} = siteDataJson

Header.defaultProps = {
    title
}

export default Header
