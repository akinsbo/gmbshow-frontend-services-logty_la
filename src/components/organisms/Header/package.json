{
    "name": "header",
    "version": "1.0.0",
    "description": "Header component",
    "main": "Header.tsx",
    "scripts": {
        "test": "jest"
    },
    "repository": {
        "type": "git",
        "url": "https://github.com/akinsbo/gmbshow-frontend-services-logty_la/src/components/atoms/Header"
    },
    "keywords": [
        "header",
        "component",
        "material"
    ],
    "author": "Olaolu Akinsete",
    "license": "ISC"
}