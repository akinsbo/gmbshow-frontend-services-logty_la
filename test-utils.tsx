import React from 'react'
import { render, queries } from '@testing-library/react'
import { ThemeProvider } from '@material-ui/styles'
import { createMuiTheme } from '@material-ui/core/styles'
import { I18nextProvider } from 'react-i18next'
import defaultStrings from './i18n/en-x-default'
import * as customQueries from './custom-queries'
import grey from '@material-ui/core/colors/grey'
import orange from '@material-ui/core/colors/orange'
import purple from '@material-ui/core/colors/purple'
import yellow from '@material-ui/core/colors/yellow'
// A theme with custom primary and secondary color.
// It's optional.
const theme = createMuiTheme({
    palette: {
        primary: {
            dark: purple[900],
            light: orange[500],
            main: grey[700]
        },
        secondary: {
            dark: yellow[700],
            light: purple[300],
            main: yellow[500]
        },
    }
})

const AllTheProviders = ({ children }: any) =>
    <ThemeProvider theme={theme} >
        <I18nextProvider i18n={defaultStrings} >
            {children}
        </I18nextProvider>
    </ThemeProvider >


const customRender = (ui: any, options = {}) =>
    render(ui, { wrapper: AllTheProviders, queries: { ...queries, ...customQueries }, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }