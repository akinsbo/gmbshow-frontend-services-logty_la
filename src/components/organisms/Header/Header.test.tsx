import * as React from "react"
import { render } from '../../../../test-utils'
import Header from "./Header"

describe("Header", () => {
  it("renders correctly", () => {
    const { container } = render(
      <Header />
    )
    expect(container).toMatchSnapshot()
  })
})