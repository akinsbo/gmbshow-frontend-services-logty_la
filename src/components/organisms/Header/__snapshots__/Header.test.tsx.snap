// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div>
  <div
    class="header makeStyles-grow-1"
    data-cy="header"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="Open drawer"
          class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <h6
          class="MuiTypography-root makeStyles-titleStyle-4 MuiTypography-h6 MuiTypography-colorInherit MuiTypography-noWrap"
        >
          title
        </h6>
        <div
          class="makeStyles-grow-1"
        />
        <div
          class="makeStyles-sectionDesktop-6"
        />
        <div
          class="makeStyles-sectionMobile-7"
        />
      </div>
    </header>
  </div>
</div>
`;
